tests/utils.py

```diff
@@ -235,6 +235,52 @@ def prepend_tmp(path):
 )
 
 
+def run_emulated(simulation: Simulation, task_name: str = None) -> SimulationData:
+    """Emulates a simulation run."""
+
+    def make_data(coords: dict, data_array_type: type, is_complex: bool = False) -> "data_type":
+        """make a random DataArray out of supplied coordinates and data_type."""
+        data_shape = [len(coords[k]) for k in data_array_type.__slots__]
+        data = np.random.random(data_shape)
+        data = (1 + 1j) * data if is_complex else data
+        return data_array_type(data, coords=coords)
+
+    def make_field_data(monitor: FieldMonitor) -> FieldData:
+        """make a random FieldData from a FieldMonitor."""
+        field_cmps = {}
+        coords = {"f": list(monitor.freqs)}
+        rmin, rmax = simulation.bounds
+        for dim, pos_min, pos_max in zip("xyz", rmin, rmax):
+            coords[dim] = np.linspace(pos_min, pos_max, 101)
```

Comment:
For this to be consistent with what will actually be returned by the solver, the `coords` should be given by something like `simulation.discretize(monitor, extend=True).yee.E.x` for the Ex component, etc.

Well, there are some further considerations:
- if the monitor size is 0 along a certain dimension, the coord is just the center along that axis
- some monitors allow for spatial downsampling (record every `interval` spatial steps)

This is making me wonder if we shouldn't have something like a `Monitor.discretize(grid)` method instead of handling those things on the backend.

But yeah just generally it would be good to have this function return data with shapes matching what will actually be returned by the solver as closely as possible.

Reply:
yea, I omitted this out of laziness, but planned to add it in if this function sees more use.  I've added it in now:
```py
    def make_field_data(monitor: FieldMonitor) -> FieldData:
        """make a random FieldData from a FieldMonitor."""
        field_cmps = {}
        coords = {"f": list(monitor.freqs)}
        rmin, rmax = simulation.bounds
        for dim, pos_min, pos_max in zip("xyz", rmin, rmax):
            coords[dim] = np.linspace(pos_min, pos_max, 101)

        grid = simulation.discretize(monitor, extend=True)

        for field_name in monitor.fields:
            spatial_coords_dict = grid[field_name].dict()

            for axis, dim in enumerate('xyz'):
                if monitor.size[axis] == 0:
                    coords[dim] = [monitor.center[axis]]
                else:
                    coords[dim] = np.array(spatial_coords_dict[dim])

            field_cmps[field_name] = make_data(
                coords=coords, data_array_type=ScalarFieldDataArray, is_complex=True
            )

        return FieldData(monitor=monitor, **field_cmps)
```